File: Xamarin/Activity3-3/Activity3-3/JSONAdapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Org.Json;
using Android.Util;

namespace Activity3_3
{
    class JSONAdapter : BaseAdapter
    {

        Activity activity;
        JSONArray data;

        public JSONAdapter(Activity activity, JSONArray data)
        {
            this.activity = activity;
            this.data = data;
        }

        public override int Count
        {
            get
            {
                int length = 0;
                try
                {
                    length = data.Length();
                }
                catch(Exception e)
                {
                    Log.Debug("EXCEPTION", e.Message);
                }
                return length;
            }
        }

        public override Java.Lang.Object GetItem(int position)
        {
            JSONObject toReturn = null;
            try
            {
                toReturn = data.GetJSONObject(position);
            }
            catch(Exception e)
            {
                Log.Debug("EXCEPTION", e.Message);
            }
            return toReturn;
        }

        public override long GetItemId(int position)
        {
            return position;
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            if(convertView == null)
            {
                convertView = activity.LayoutInflater.Inflate(Resource.Layout.layout1, null);
            }

            string title = "";
            string body = "";

            try
            {
                title = data.GetJSONObject(position).GetString("title");
                body = data.GetJSONObject(position).GetString("body");
            }
            catch(Exception e)
            {
                Log.Debug("EXCEPTION", e.Message);
            }

            TextView large = convertView.FindViewById<TextView>(Resource.Id.textView1);
            TextView small = convertView.FindViewById<TextView>(Resource.Id.textView2);

            large.Text = title;
            small.Text = body;

            return convertView;
        }
    }
}

File: AndroidStudio/Activity2-1/app/src/main/java/mx/itesm/activity2_1/JSONAdapter.java
package mx.itesm.activity2_1;

import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by forne on 18/03/2017.
 */

public class JSONAdapter extends BaseAdapter {
    private JSONArray data;
    private Activity activity;

    public JSONAdapter(JSONArray data, Activity activity){
        this.data = data;
        this.activity = activity;
    }

    @Override
    public int getCount() {
        return data.length();
    }

    @Override
    public Object getItem(int position) {
        try{
            return data.getJSONObject(position);
        }catch(JSONException joe){
            joe.printStackTrace();
        }
        return null;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if(convertView == null){
            convertView = activity.getLayoutInflater().inflate(R.layout.row,null);
        }
        TextView name = (TextView) convertView.findViewById(R.id.name);
        TextView hobby = (TextView) convertView.findViewById(R.id.hobby);

        try{
            JSONObject friend = data.getJSONObject(position);
            name.setText(friend.getString("name"));
            hobby.setText(friend.getString("hobby"));
        }catch(JSONException joe){
            joe.printStackTrace();
        }
        return convertView;
    }
}
